### PR TITLE
Fixed #31552 -- Added support for LZMA and XZ fixtures to loaddata.

### PR DESCRIPTION
**Pull request by pauloxnet**
https://code.djangoproject.com/ticket/31552

### CODE REVIEW

File: docs/ref/django-admin.txt [hunk 1]
```diff
@@ -601,12 +601,13 @@ The :djadmin:`dumpdata` command can be used to generate input for ``loaddata``.
...
-Fixtures may be compressed in ``zip``, ``gz``, or ``bz2`` format. For example::
+Fixtures may be compressed in ``zip``, ``gz``, ``bz2`` or ``xz`` format.
+For example::
 
     django-admin loaddata mydata.json
 
-would look for any of ``mydata.json``, ``mydata.json.zip``,
-``mydata.json.gz``, or ``mydata.json.bz2``. The first file contained within a
+would look for any of ``mydata.json``, ``mydata.json.zip``, ``mydata.json.gz``,
+``mydata.json.bz2`` or ``mydata.json.xz``. The first file contained within a
```
Comment (felixxm) [inline docs/ref/django-admin.txt:610]: Serial comma should stay before 'or' (see https://en.wikipedia.org/wiki/Serial_comma).

File: django/core/management/commands/loaddata.py [hunk 2]
```diff
@@ -97,6 +103,8 @@ def loaddata(self, fixture_labels):
         }
         if has_bz2:
             self.compression_formats['bz2'] = (bz2.BZ2File, 'r')
+        if has_lzma:
+            self.compression_formats['xz'] = (lzma.LZMAFile, 'r')
```
Comment (felixxm) [inline django/core/management/commands/loaddata.py:107]: I think we should add also `lzma` format.

File: docs/ref/django-admin.txt [hunk 3]
```diff
@@ -601,12 +601,13 @@ The :djadmin:`dumpdata` command can be used to generate input for ``loaddata``.
 Compressed fixtures
 ~~~~~~~~~~~~~~~~~~~
 
-Fixtures may be compressed in ``zip``, ``gz``, or ``bz2`` format. For example::
+Fixtures may be compressed in ``zip``, ``gz``, ``bz2`` or ``xz`` format.
```
Comment (felixxm) [inline docs/ref/django-admin.txt:604]: Serial comma should stay before 'or' (see https://en.wikipedia.org/wiki/Serial_comma).

File: tests/fixtures/tests.py [hunk 4]
```diff
@@ -559,6 +565,13 @@ def test_compressed_loading_bz2(self):
             '<Article: WoW subscribers now outnumber readers>',
         ])
 
+    @unittest.skipUnless(HAS_LZMA, 'No lzma library detected.')
+    def test_compressed_loading_lzma(self):
+        management.call_command('loaddata', 'fixture5.json.xz', verbosity=0)
+        self.assertQuerysetEqual(Article.objects.all(), [
+            '<Article: WoW subscribers now outnumber readers>',
+        ])
```
Comment (felixxm) [inline tests/fixtures/tests.py:573]: Please add test also for `.lzma` archive.

File: docs/ref/django-admin.txt (hunk 1 above)
Comment (felixxm) [reply docs/ref/django-admin.txt:610]: `versionchanged`  annotation is missing, e.g.:
```
.. versionchanged:: 3.2

    Support for XZ archives (``.xz``) and LZMA archives (``.lzma``) was added.
```

**Comment by pauloxnet** [comment]
@felixxm thanks for your review int he updated commit I added support for LZMA archives , fixes documentation issues and added a section in 3.2 release notes.

**Comment by felixxm** [comment]
@pauloxnet Thanks :+1:

**Comment by pauloxnet** [comment]
@felixxm Thanks for your review and merge. 👌

**Pull request merged**